src/App.jsx

```diff
@@ -1,248 +1,156 @@
-                    </span>
-                  </a>
-                </span>
-              </th>
-            </tr>
-          </thead>
-
-          <tbody>
-            <tr data-cy="Product">
-              <td className="has-text-weight-bold" data-cy="ProductId">
-                1
-              </td>
-
-              <td data-cy="ProductName">Milk</td>
-              <td data-cy="ProductCategory">🍺 - Drinks</td>
-
-              <td
-                data-cy="ProductUser"
-                className="has-text-link"
-              >
-                Max
-              </td>
-            </tr>
-
-            <tr data-cy="Product">
-              <td className="has-text-weight-bold" data-cy="ProductId">
-                2
-              </td>
-
-              <td data-cy="ProductName">Bread</td>
-              <td data-cy="ProductCategory">🍞 - Grocery</td>
-
-              <td
-                data-cy="ProductUser"
-                className="has-text-danger"
-              >
-                Anna
-              </td>
-            </tr>
-
-            <tr data-cy="Product">
-              <td className="has-text-weight-bold" data-cy="ProductId">
-                3
-              </td>
-
-              <td data-cy="ProductName">iPhone</td>
-              <td data-cy="ProductCategory">💻 - Electronics</td>
-
-              <td
-                data-cy="ProductUser"
-                className="has-text-link"
+import usersFromServer from './api/users';
+import categoriesFromServer from './api/categories';
+import productsFromServer from './api/products';
+import { Users } from './components/Users/Users';
+import { InputField } from './components/InputField/InputField';
+import { Category } from './components/Category/Category';
+import { Products } from './components/Products/Products';
+
+const products = productsFromServer.map((product) => {
+  const category = categoriesFromServer
+    .find(categ => product.categoryId === categ.id);
+
+  const user = usersFromServer
+    .find(human => category.ownerId === human.id);
+
+  return {
+    product,
+    category,
+    user,
+  };
+});
+
+const filterProducts = (goods, query, categories, user) => {
+  let goodsArray = [...goods];
+
+  if (query) {
+    goodsArray = goodsArray.filter(good => good.product.name
+      .toLowerCase().includes(query.toLowerCase().trim()));
+  }
+
+  if (categories.length !== 0) {
+    goodsArray = goodsArray
+      .filter(good => categories.includes(good.category.title));
+  }
+
+  if (user) {
+    goodsArray = goodsArray.filter(good => good.user.name === user);
+  }
+
+  return goodsArray;
+};
+
+const sortProducts = (productsArray, sortType) => {
+  const array = [...productsArray];
+
+  if (Object.keys(sortType).length === 0) {
+    return array;
+  }
+
+  if (sortType.ID) {
+    if (sortType.ID === 'ASC') {
+      array.sort((a, b) => a.product.id - b.product.id);
+    } else {
+      array.sort((a, b) => b.product.id - a.product.id);
```

Comment:
You can put the if...else (or even better - ternary operator) inside the sort callback